instrumentation/opentelemetry-instrumentation-aio-pika/src/opentelemetry/instrumentation/aio_pika/span_builder.py

```diff
@@ -64,20 +63,13 @@ def set_message(self, message: AbstractMessage):
             self._attributes[SpanAttributes.MESSAGING_CONVERSATION_ID] = properties.correlation_id
 
     def build(self) -> Optional[Span]:
-        if context.get_value('suppress_instrumentation') or context.get_value(context._SUPPRESS_INSTRUMENTATION_KEY):
+        if not is_instrumentation_enabled():
             return None
-        assert self._kind, 'kind must be configured.'
-        assert self._destination, 'destination must be configured.'
-        tracer = trace.get_tracer(__name__, __version__, self.TRACER_PROVIDER)
-        span = tracer.start_span(self._generate_span_name(), kind=self._kind)
-        if not span.is_recording():
-            return span
-        for attribute_name, attribute_value in self._attributes.items():
-            span.set_attribute(attribute_name, attribute_value)
         if self._operation:
-            span.set_attribute(SpanAttributes.MESSAGING_OPERATION, self._operation.value)
+            self._attributes[SpanAttributes.MESSAGING_OPERATION] = self._operation.value
         else:
-            span.set_attribute(SpanAttributes.MESSAGING_TEMP_DESTINATION, True)
+            self._attributes[SpanAttributes.MESSAGING_TEMP_DESTINATION] = True
```

Comment:
You shouldn't access `self._attributes` like that. This sets the attributes for the tracer provider (which means they're part of the "process" information of the span).
Also, it's not thread-safe
Create the span and use `set_attribute` :)